File: rc-application/src/main/java/com/transkript/reportcard/business/service/ClassLevelSubService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.ClassLevelSubDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.ClassLevelSub;

import java.util.List;

public interface ClassLevelSubService {
    EntityResponse addClassLevelSub(ClassLevelSubDto classLevelSubDto);

    List<ClassLevelSubDto> getClassLevelSubs();

    List<ClassLevelSubDto> getClassLevelSubsByClassLevel(Long levelId);

    ClassLevelSubDto getClassLevelSub(Long id);

    EntityResponse updateClassLevelSub(Long id, ClassLevelSubDto classLevelSubDto);

    String deleteClassLevelSub(Long id);

    ClassLevelSub getClassLevelSubEntity(Long id);

}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/ClassLevelSubService.kt
package com.transkript.reportcard.business.service

import com.transkript.reportcard.api.dto.ClassLevelSubDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.ClassLevelSub

interface ClassLevelSubService {
    fun addClassLevelSub(classLevelSubDto: ClassLevelSubDto?): EntityResponse?
    val classLevelSubs: List<ClassLevelSubDto?>?
    fun getClassLevelSubsByClassLevel(levelId: Long?): List<ClassLevelSubDto?>?
    fun getClassLevelSub(id: Long?): ClassLevelSubDto?
    fun updateClassLevelSub(id: Long?, classLevelSubDto: ClassLevelSubDto?): EntityResponse?
    fun deleteClassLevelSub(id: Long?)
    fun getClassLevelSubEntity(id: Long?): ClassLevelSub?
}
